frontend/web/components/pages/EnvironmentSettingsPage.tsx

```diff
@@ -0,0 +1,891 @@
+                            >
+                              {isSaving ? 'Updating' : 'Update'}
+                            </Button>
+                          </div>
+                        </form>
+                      </div>
+                      <hr className='py-0 my-4' />
+                      <div className='col-md-8 mt-4'>
+                        <Setting
+                          onChange={(value) => updateCurrentEnv({
+                            banner_text: value ? `${currentEnv?.name} Environment` : null
+                          }, true)}
+                          checked={typeof currentEnv?.banner_text === 'string'}
+                          title={'Environment Banner'}
+                          description={
+                            <div>
+                              This will show a banner whenever you view its
+                              pages.
+                              <br />
+                              This is generally used to warn people that they
+                              are viewing and editing a sensitive environment.
+                            </div>
+                          }
+                        />
+                        {typeof currentEnv?.banner_text === 'string' && (
+                          <Row className='mt-4 flex-nowrap'>
+                            <Input
+                              placeholder='Banner text'
+                              value={currentEnv?.banner_text}
+                              onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                                const bannerText = Utils.safeParseEventValue(e)
+                                updateCurrentEnv({ banner_text: bannerText }, false)
+                              }}
+                              className='full-width'
+                            />
+                            <div className='ml-2'>
+                              <ColourSelect
+                                value={currentEnv?.banner_colour || ''}
+                                onChange={(banner_colour) =>
+                                  updateCurrentEnv({ banner_colour }, false)
+                                }
+                              />
+                            </div>
+                            <Button onClick={() => saveEnv()} size='small'>
+                              Save
+                            </Button>
+                          </Row>
+                        )}
+                      </div>
+                      {Utils.getFlagsmithHasFeature('feature_versioning') && (
+                        <div>
+                          <div className='col-md-8 mt-4'>
+                            {currentEnv?.use_v2_feature_versioning === false && (
+                              <EnvironmentVersioningListener
+                                // TODO: Should be listening on Id or API_KEY ?
+                                id={env.api_key}
```

Comment:
Not sure here what should be the key

Reply:
It's the api key, this component polls the environment when enabling feature versioning (an upcoming feature).

The process is async, saving the environment doesn't process right away so we listen for a change in data and respond to it.

Reply:
Ok! Leaving it as of for now then, i'll remove the comment